src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizExerciseResource.java

```diff
@@ -230,13 +285,179 @@ public QuizExerciseResource(QuizExerciseRepository quizExerciseRepository, Parti
+                        if ((question.isInvalid() && !originalQuestion.isInvalid()) ||
+                            !question.getScoringType().equals(originalQuestion.getScoringType())) {
+                            updateResultsAndStatistics = true;
+                        }
+
+                        if (question instanceof MultipleChoiceQuestion) {
+                            //find added Answers, which are not allowed to be added
+                            Set<AnswerOption> notAllowedAddedAnswers = new HashSet<AnswerOption>();
+                            for (AnswerOption answer : (((MultipleChoiceQuestion) question).getAnswerOptions())) {
+                                if (((MultipleChoiceQuestion) originalQuestion).getAnswerOptions().contains(answer)) {
+                                    //find original answer
+                                    for (AnswerOption originalAnswer : ((MultipleChoiceQuestion) originalQuestion).getAnswerOptions()) {
+                                        if (answer.getId().equals(originalAnswer.getId())) {
+                                            //reset invalid answer if it already set to true (it's not possible to set an answer valid again)
+                                            answer.setInvalid(answer.isInvalid() || originalAnswer.isInvalid());
+
+                                            // check if an answer is  set invalid or if the correctness has changed
+                                            if ((answer.isInvalid() && !originalAnswer.isInvalid() && !question.isInvalid())||
+                                                (!(answer.isIsCorrect().equals(originalAnswer.isIsCorrect())))) {
+                                                updateResultsAndStatistics = true;
+                                            }
+                                        }
+
+                                    }
+                                } else {
+                                    //note the added Answers
+                                    notAllowedAddedAnswers.add(answer);
+                                }
+                            }
+                            //remove the added Answers
+                            ((MultipleChoiceQuestion) question).getAnswerOptions().removeAll(notAllowedAddedAnswers);
+
+                            // check if an answer was deleted
+                            if (((MultipleChoiceQuestion) question).getAnswerOptions().size() <
+                                ((MultipleChoiceQuestion) originalQuestion).getAnswerOptions().size()) {
+                                updateResultsAndStatistics = true;
+                            }
+                        }
+                    }
+                }
+            } else {
+                // question is added (not allowed)
+                addedQuestions.add(question);
+            }
+        }
```

Comment:
The whole code above is quite complex. Can you please try to restructure it a bit more. It might e.g. be good to extract methods for certain parts. Definitely add more documentation, otherwise it's a nightmare for other developers to maintain this code!